lib/beacon/loader/snippets.ex

```diff
@@ -1,18 +1,21 @@
-defmodule Beacon.Loader.SnippetModuleLoader do
+defmodule Beacon.Loader.Snippets do
   @moduledoc false
 
   alias Beacon.Loader
 
-  def load_helpers(_site, [] = _helpers) do
-    :skip
+  def module_name(site), do: Loader.module_name(site, "Snippets")
+
+  def build_ast(site, snippets) do
+    module = module_name(site)
+    functions = Enum.map(snippets, &helper/1)
+    render(module, functions)
   end
 
-  def load_helpers(site, helpers) do
-    module_name = Loader.snippet_helpers_module_for_site(site)
-    functions = Enum.map(helpers, &helper/1)
-    ast = render(module_name, functions)
-    :ok = Loader.reload_module!(module_name, ast)
-    {:ok, ast}
+  defp render(module_name, [] = _functions) do
+    quote do
+      defmodule unquote(module_name) do
+      end
+    end
```

Comment:
Interesting!  What happens when we compile `(unquote_splicing([]))`?

Reply:
It returns an empty block but I think that's still valid so we could have a single render/2 here